### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by ebassi**
Not ready to merge, yet, but the first two commits can be cherry picked.

### CODE REVIEW

**Comment by alexlarsson** [comment]
Cherry-picket the first two

File: gthree/gthreeresource.h [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (alexlarsson) [inline gthree/gthreeresource.h]: Uhm, any particular reasons this one is LGPL as opposed to MIT?

Comment (ebassi) [reply gthree/gthreeresource.h]: No, I just slipped on the copy-pasta.

**Comment by alexlarsson** [comment]
closing this for now

**Pull request closed without merging**